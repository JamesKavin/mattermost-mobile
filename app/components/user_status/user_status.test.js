// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {General} from '@constants';
import {renderWithIntlAndTheme} from '@test/intl-test-helper';

import UserStatus from './';

describe('UserStatus', () => {
    const baseProps = {
        size: 32,
    };

    test('should match snapshot, should default to offline status', () => {
        const wrapper = renderWithIntlAndTheme(
            <UserStatus {...baseProps}/>,
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('should match snapshot, away status', () => {
        const wrapper = renderWithIntlAndTheme(
            <UserStatus
                {...baseProps}
                status={General.AWAY}
            />,
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('should match snapshot, dnd status', () => {
        const wrapper = renderWithIntlAndTheme(
            <UserStatus
                {...baseProps}
                status={General.DND}
            />,
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('should match snapshot, online status', () => {
        const wrapper = renderWithIntlAndTheme(
            <UserStatus
                {...baseProps}
                status={General.ONLINE}
            />,
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
