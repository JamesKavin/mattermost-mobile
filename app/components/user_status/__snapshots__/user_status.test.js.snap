// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserStatus should match snapshot, away status 1`] = `
<Icon
  name="clock"
  style={
    {
      "color": "#ffbc1f",
      "fontSize": 32,
    }
  }
  testID="user_status.indicator.away"
/>
`;

exports[`UserStatus should match snapshot, dnd status 1`] = `
<Icon
  name="minus-circle"
  style={
    {
      "color": "#d24b4e",
      "fontSize": 32,
    }
  }
  testID="user_status.indicator.dnd"
/>
`;

exports[`UserStatus should match snapshot, online status 1`] = `
<Icon
  name="check-circle"
  style={
    {
      "color": "#3db887",
      "fontSize": 32,
    }
  }
  testID="user_status.indicator.online"
/>
`;

exports[`UserStatus should match snapshot, should default to offline status 1`] = `
<Icon
  name="circle-outline"
  style={
    {
      "color": "rgba(184,184,184,0.64)",
      "fontSize": 32,
    }
  }
  testID="user_status.indicator.offline"
/>
`;
